sklearn/linear_model/cd_fast.pyx

```diff
@@ -232,23 +232,20 @@ def enet_coordinate_descent(floating[::1] w,
                 # tmp = (X[:,ii]*R).sum()
                 tmp = dot(n_samples, &X[0, ii], 1, &R[0], 1)
 
-                if positive and tmp < 0:
+                if (positive or -alpha <= tmp) and (tmp <= alpha):
                     w[ii] = 0.0
                 else:
                     w[ii] = (fsign(tmp) * fmax(fabs(tmp) - alpha, 0)
                              / (norm_cols_X[ii] + beta))
 
-                if w[ii] != 0.0:
```

Comment:
it's very common that w[ii] will be zero after soft thresholding so I suspect this slows up things.

Reply:
Also not following this either unfortunately. Given the condition in the `if`, the `else` case can only make `w[ii] != 0`. So the concern raised shouldn't actually occur. Feel free to clarify if I'm missing something though.

Reply:
Exactly. This is being run in the `else`, which does not set `w[ii]` to `0.0`.

Reply:
This is not in the "else", it's after it. Also, the "else" contains ``fmax(fabs(tmp) - alpha, 0)`` which is the soft-thresholding operation, so it's "likely" to result in 0.